OpenRA.Mods.Common/Traits/World/Pathfinder/PathCacheStorage.cs

```diff
@@ -0,0 +1,75 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class PathCacheStorage : ICacheStorage<List<CPos>>
+	{
+		class CachedPath
+		{
+			public List<CPos> Result;
+			public int Tick;
+		}
+
+		const int MaxPathAge = 50;
+		readonly IWorld world;
+		Dictionary<string, CachedPath> cachedPaths = new Dictionary<string, CachedPath>(1000);
+
+		public PathCacheStorage(IWorld world)
+		{
+			this.world = world;
+		}
+
+		public void Remove(string key)
+		{
+			cachedPaths.Remove(key);
+		}
+
+		public void Store(string key, List<CPos> data)
+		{
+			// Eventually clean up the cachedPaths dictionary
+			if (cachedPaths.Count >= 1000)
+				foreach (var cachedPath in cachedPaths.Where(p => IsExpired(p.Value)).ToList())
```

Comment:
Yes I have. I have created a Unit test (that you will be able to execute) which benchmarks the act of removing all the paths from the list, and on a I3 processor it takes 7 ms to execute those 1000 checks and removals.

I think we can deal with that.